### PR TITLE
Rework collision geometries for bases

### PR DESCRIPTION
**Pull request by raybjork**
This PR does two things
- removes a warning that was occurring when no geometry was configured for a base
- if no frame is specified for the base the kinematic wrapper wrapping it will assign it a point collision geometry

### CODE REVIEW

File: spatialmath/geometry_utils.go [hunk 1]
```diff
@@ -81,6 +81,24 @@ func PlaneNormal(p0, p1, p2 r3.Vector) r3.Vector {
...
+		r += r3.Vector{X: g.halfSize[0], Y: g.halfSize[1], Z: g.halfSize[2]}.Norm()
+	case *sphere:
+		r += g.radius
+	case *capsule:
+		r += g.length / 2
+	case *point:
+	default:
+		return nil, ErrGeometryTypeUnsupported
+	}
+	return NewSphere(NewZeroPose(), r, geometry.Label())
```
Comment (biotinker) [inline spatialmath/geometry_utils.go:99]: Is there a reason not to have this be `geometry.Pose()` and remove the pose norm from `r`?

File: components/base/kinematicbase/differentialDrive.go [hunk 2]
```diff
@@ -54,13 +54,21 @@ func wrapWithDifferentialDriveKinematics(
 	if len(geometries) > 0 {
 		geometry = geometries[0]
 	}
-	ddk.executionFrame, err = referenceframe.New2DMobileModelFrame(b.Name().ShortName(), limits, geometry)
+	sphere, err := spatialmath.BoundingSphere(geometry)
+	if err != nil {
+		// default to using a point to represent the base for the purposes of collision handling rather than erroring
+		logger.Warnf("base %s not configured with a geometry, defaulting to considering it a point mass." +
```
Comment (randhid) [inline components/base/kinematicbase/differentialDrive.go:60]: Pretty long, any way to make it snappier?

File: components/base/wheeled/wheeled_base.go [hunk 3]
```diff
@@ -122,11 +121,14 @@ func (wb *wheeledBase) Reconfigure(ctx context.Context, deps resource.Dependenci
...
-	geometries, err := kinematicbase.CollisionGeometry(conf.Frame)
-	if err != nil {
-		wb.logger.Warnf("base %v %s", wb.Name(), err.Error())
+	wb.geometries = []spatialmath.Geometry{}
+	if conf.Frame != nil {
+		frame, err := conf.Frame.ParseConfig()
+		if err != nil {
+			return err
+		}
+		wb.geometries = append(wb.geometries, frame.Geometry())
```
Comment (randhid) [inline components/base/wheeled/wheeled_base.go:130]: Re-Checking fake base didn't need an update.

File: spatialmath/geometry_utils.go [hunk 4]
```diff
@@ -81,6 +81,24 @@ func PlaneNormal(p0, p1, p2 r3.Vector) r3.Vector {
 	return p1.Sub(p0).Cross(p2.Sub(p0)).Normalize()
 }
 
+// Bounding sphere returns a spherical geometry centered on the point (0, 0, 0) that will encompass the given geometry
+// if it were to be rotated 360 degrees about the Z axis.  The label of the new geometry is inherited from the given one.
+func BoundingSphere(geometry Geometry) (Geometry, error) {
```
Comment (randhid) [inline spatialmath/geometry_utils.go:86]: Do you not want ever to expand this function to return other geometry types? It's contained now in the kinematic base package so it fine, but it seems weird (long-term, not now) to input a config for a box in your base, or code `NewBox` in for go and get a sphere back.

Nothing to do here, just a question.

File: spatialmath/geometry_utils.go [hunk 5]
```diff
@@ -81,6 +81,24 @@ func PlaneNormal(p0, p1, p2 r3.Vector) r3.Vector {
...
+	switch g := geometry.(type) {
+	case *box:
+		r += r3.Vector{X: g.halfSize[0], Y: g.halfSize[1], Z: g.halfSize[2]}.Norm()
+	case *sphere:
+		r += g.radius
+	case *capsule:
+		r += g.length / 2
+	case *point:
+	default:
+		return nil, ErrGeometryTypeUnsupported
```
Comment (randhid) [inline spatialmath/geometry_utils.go:97]: does this error need to be exported?

**Review by randhid — APPROVED** [review]
LGTM with a few nits and one question.

File: spatialmath/geometry_utils.go (hunk 5 above)
Comment (raybjork) [reply spatialmath/geometry_utils.go:97]: I think with this change we no longer need to.  I will try to unexport

File: spatialmath/geometry_utils.go (hunk 4 above)
Comment (raybjork) [reply spatialmath/geometry_utils.go:86]: I'm not sure I understand the question.  You mean write another function `BoundingPrism`? We could certainly do that, there just hasnt been a need to yet.

File: components/base/kinematicbase/differentialDrive.go (hunk 2 above)
Comment (raybjork) [reply components/base/kinematicbase/differentialDrive.go:60]: You're right I will try and channel Hemingway on the rewrite

File: components/base/wheeled/wheeled_base.go (hunk 3 above)
Comment (biotinker) [reply components/base/wheeled/wheeled_base.go:130]: Can confirm

**Review by biotinker — APPROVED** [review]
LGTM

File: spatialmath/geometry_utils.go (hunk 4 above)
Comment (randhid) [reply spatialmath/geometry_utils.go:86]: The flow of adding a box geometry to your robot config and this function outputting a sphere no matter what you input is kind of weird/funny. But you're right. There is no need, currently, to give back other geometry types. But I would keep that in mind as you expose more motion/nav capabilities. 😄

Comment (raybjork) [reply spatialmath/geometry_utils.go:86]: So this is only used in the differential drive kinematic base, which does not consider orientation in creating its plans.  The only safe thing to do imo is create a sphere to use for collision checking.  Technically the underlying base itself will always have the user configured geometry

And for PTGs we now use the user specified geometry

**Pull request merged**